src/test/java/org/opensearch/neuralsearch/search/query/HybridCollectorManagerTests.java

```diff
@@ -1169,4 +1169,41 @@ public void testScrollWithHybridQuery_thenFail() {
             illegalArgumentException.getMessage()
         );
     }
+
+    @SneakyThrows
+    public void testCreateCollectorManager_whenPaginationDepthIsEqualToNullAndFromIsGreaterThanZero_thenFail() {
```

Comment:
Do we also have a test when a pagination depth is provided to verify the results?

Reply:
We already have that integ test written above in this file. Code Ref: https://github.com/opensearch-project/neural-search/blob/main/src/test/java/org/opensearch/neuralsearch/search/query/HybridCollectorManagerTests.java#L1105